src/Sylius/Bundle/ApiBundle/Controller/GetStatisticsAction.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Controller;
+
+use Sylius\Bundle\ApiBundle\Query\GetStatistics;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Messenger\Exception\HandlerFailedException;
+use Symfony\Component\Messenger\HandleTrait;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+
+final class GetStatisticsAction
+{
+    use HandleTrait;
+
+    /** @var array<array{queryParameter: string, message: string}> */
+    private array $violations = [];
+
+    /** @var array<string> */
+    private array $requiredParameters = [
```

Comment:
General naming idea, that it should be called queryParameters, not just parameters.

```suggestion
    private array $requiredQueryParameters = [
```